Modules/Core/Common/test/itkShapedImageNeighborhoodRangeGTest.cxx

```diff
@@ -544,7 +544,7 @@ TEST(ShapedImageNeighborhoodRange, IteratorRetrievesSamePixelValuesAsConstNeighb
 
     itk::SizeValueType i = 0;
 
-    for (const auto pixel : range)
+    for (const auto & pixel : range)
```

Comment:
The `auto` is setting the pixel type to a `PixelProxy`, which is great in some circumstances, but in this case I'd venture that a simple `PixelType` is what is intended.  Especially because `PixelType` is `unsigned char` here, I might instead go with
```c++
for (const PixelType pixel : range)
```

Reply:
In general I think the syntax `for (const auto value : range)` should be fully supported, especially for generic code. But if that causes warnings that need to be addressed, I'm OK with both `for (const PixelType pixel...)` and `for (const auto & pixel...)`.

Reply:
Changed to `const PixelType pixel` in 4a75ae8.